src/TulsiGenerator/Bazel/tulsi/tulsi_aspects.bzl

```diff
@@ -557,6 +564,7 @@ def _tulsi_sources_aspect(target, ctx):
       bridging_header=_collect_first_file(rule_attr, 'bridging_header'),
       compiler_defines=_extract_compiler_defines(ctx),
       enable_modules=_get_opt_attr(rule_attr, 'enable_modules'),
+      module_name=_get_opt_attr(rule_attr, 'module_name') or _get_module_name(str(target.label)),
```

Comment:
The fallback to the target label would be applied  to any non-swift and non-objc target as well. We should restrict it to targets that have objc or swift providers.